test/e2e/infoblox.go

```diff
@@ -0,0 +1,296 @@
+		Infoblox: &operatorv1alpha1.ExternalDNSInfobloxProviderOptions{
+			Credentials: operatorv1alpha1.SecretReference{
+				Name: credsSecret.Name,
+			},
+			GridHost:    h.gridHost,
+			WAPIPort:    wapiPort,
+			WAPIVersion: defaultWAPIVersion,
+		},
+	}
+	return resource
+}
+
+func (h *infobloxTestHelper) prepareConfigurations(kubeClient client.Client) error {
+	configDir := os.Getenv(infobloxGridConfigDirEnvVar)
+	if configDir != "" {
+		host, err := ioutil.ReadFile(configDir + "/" + defaultHostFilename)
+		if err != nil {
+			return fmt.Errorf("failed to read grid host from file: %w", err)
+		}
+		username, err := ioutil.ReadFile(configDir + "/" + defaultUsernameFilename)
+		if err != nil {
+			return fmt.Errorf("failed to read wapi username from file: %w", err)
+		}
+		password, err := ioutil.ReadFile(configDir + "/" + defaultPasswordFilename)
+		if err != nil {
+			return fmt.Errorf("failed to read wapi password from file: %w", err)
+		}
+		h.gridHost = string(host)
+		h.wapiUsername = string(username)
+		h.wapiPassword = string(password)
+	} else {
+		h.gridHost = mustGetEnv(infobloxGridHostEnvVar)
+		h.wapiUsername = mustGetEnv(infobloxWAPIUsernameEnvVar)
+		h.wapiPassword = mustGetEnv(infobloxWAPIPasswordEnvVar)
+	}
+
+	return h.trustGridTLSCert()
+}
+
+// trustGridTLSCert instructs the operator to trust Grid Master's self signed TLS certificate.
+func (h *infobloxTestHelper) trustGridTLSCert() error {
+	// get Grid's TLS certificate as raw PEM encoded data
+	certRaw, err := readServerTLSCert(net.JoinHostPort(h.gridHost, defaultWAPIPort), true)
+	if err != nil {
+		return fmt.Errorf("failed to get Grid Master's TLS certificate: %w", err)
+	}
+
+	operatorNs := "external-dns-operator"
+	trustedCAConfigMapName := fmt.Sprintf("infoblox-trustedca-%s", randomString(16))
+
+	// create a config map with the trusted CA bundle
+	trustedCAConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      trustedCAConfigMapName,
+			Namespace: operatorNs,
+		},
+		Data: map[string]string{
+			"ca-bundle.crt": string(certRaw),
+		},
+	}
+	if err := kubeClient.Create(context.TODO(), trustedCAConfigMap); err != nil {
+		return fmt.Errorf("failed to create trusted CA configmap %s/%s: %w", trustedCAConfigMap.Namespace, trustedCAConfigMap.Name, err)
+	}
+
+	// trusted CA environment variable to be injected
+	trustedCAEnvVar := corev1.EnvVar{
+		Name:  trustedCAConfigMapEnvVar,
+		Value: trustedCAConfigMapName,
+	}
+
+	// inject into subscription if there is one
+	findOperatorSubscription := func() (*olmv1alpha1.Subscription, error) {
+		list := &olmv1alpha1.SubscriptionList{}
+		if err := kubeClient.List(context.TODO(), list, client.InNamespace(operatorNs)); err != nil {
+			return nil, err
+		}
+		for i := range list.Items {
+			if strings.HasPrefix(list.Items[i].Name, "external-dns-operator") {
+				return &list.Items[i], nil
+			}
+		}
+		// best effort: return the first one
+		if len(list.Items) > 0 {
```

Comment:
It's better to error here because if there's more than one subscription then probably then we cannot safely assume later that the configmap has been correctly injected into the operator deployment.

Reply:
I agree on this one. However I noticed that the subscription created by the prow job has a generated name which is hard to guess (something like `o-abcxyz`). The listing of the subscriptions is already filtered by `external-dns-operator` namespace and after checking the must-gather I didn't see any other subscription in this namespace. Not ideal but I didn't come up with a better idea, do you have any?

Reply:
Kind of difficult to make it work both in CI and while running from local. I guess this is good enough for now.